Support building cross-architecture components on armel

echesakovMSFT opened the pull request:
@dotnet/arm32-contrib @RussKeldorph

echesakovMSFT commented:
@janvorli Can you please if this is right approach ?

RussKeldorph commented:
@dotnet/samsung

myungjoo commented on configurecompiler.cmake:
```diff
@@ -8,7 +8,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL Linux)
         # CMAKE_HOST_SYSTEM_PROCESSOR returns the value of `uname -p` on host.
         if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86_64 OR CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL amd64)
-            if(CLR_CMAKE_TARGET_ARCH STREQUAL "arm")
+            if(CLR_CMAKE_TARGET_ARCH STREQUAL "arm" OR CLR_CMAKE_TARGET_ARCH STREQUAL "armel")
```
Maybe, you will want to add "armv7l" as well if you consider softfp.

Ref: https://github.com/rpm-software-management/rpm/blob/7ffc4d17ffa8f87bd9107b5dc4d9e25daadd14ae/macros.in#L1119

janvorli replied:
@myungjoo the CLR_CMAKE_TARGET_ARCH is generated by the build.sh and we never set it to `armv7l`.

janvorli commented:
It seems we should update build.sh in this block https://github.com/dotnet/coreclr/blob/master/build.sh#L486-L510
to support armel.

alpencolt commented:
@echesakovMSFT could you explain how it will work. Is it replacement for `armelnonjit` (https://github.com/dotnet/coreclr/pull/14493)?

echesakovMSFT commented:
@alpencolt No, it's not replacement for armelnonjit. 

The reason why I added this is that I was trying to enable cross-architecture components build by default for all arm platforms and remove -crosscomponent argument from build.sh script. I noticed that command *build.sh armel cross crosscomponent* fails due to some logic missing from CMake files, so I just added this logic to have consistency among all arm/armel/arm64 platforms.

echesakovMSFT commented:
@janvorli Thanks for pointing that!

echesakovMSFT commented:
test OSX10.12 x64 Checked Innerloop Build and Test
test Windows_NT arm64 Cross Checked Innerloop Build and Test

echesakovMSFT commented:
**Windows_NT arm Cross Checked Innerloop Build and Test** and **Windows_NT arm64 Cross Checked Innerloop Build and Test** build jobs are succeeded. I am going to merge without waiting for their test jobs to finish since these changes are Linux only.
